azure-iot-hub/tests/test_iothub_configuration_manager.py

```diff
@@ -51,6 +53,65 @@ def iothub_configuration_manager():
     return iothub_configuration_manager
 
 
+@pytest.mark.describe("IoTHubConfigurationManager - .from_connection_string()")
+class TestFromConnectionString(object):
+    @pytest.mark.it(
+        "Creates an instance of ConnectionStringAuthentication and passes it to IotHubGatewayServiceAPIs constructor"
```

Comment:
Are there multiple types of valid connection strings? i.e. connection strings with optional values? Different formats? If so, this test should be parametrized to show that all valid types of strings work.